tests/components/mqtt/test_state_vacuum.py

```diff
@@ -0,0 +1,698 @@
+"""The tests for the State vacuum Mqtt platform."""
+import json
+
+import pytest
+from copy import deepcopy
+
+from homeassistant.components import mqtt, vacuum
+from homeassistant.components.mqtt import CONF_COMMAND_TOPIC, CONF_STATE_TOPIC
+from homeassistant.components.mqtt.discovery import async_start
+from homeassistant.components.mqtt.vacuum import (
+    CONF_COMPONENT, schema_state as mqttvacuum, services_to_strings)
+from homeassistant.components.mqtt.vacuum.schema_state import SERVICE_TO_STRING
+from homeassistant.components.vacuum import (
+    ATTR_BATTERY_ICON, ATTR_BATTERY_LEVEL, ATTR_FAN_SPEED, ATTR_FAN_SPEED_LIST,
+    DOMAIN, SERVICE_CLEAN_SPOT, SERVICE_LOCATE, SERVICE_PAUSE,
+    SERVICE_RETURN_TO_BASE, SERVICE_START, SERVICE_STOP, STATE_CLEANING,
+    STATE_DOCKED)
+from homeassistant.const import (
+    CONF_NAME, CONF_PLATFORM, STATE_UNAVAILABLE, STATE_UNKNOWN)
+from homeassistant.setup import async_setup_component
+
+from tests.common import (
+    MockConfigEntry, async_fire_mqtt_message, async_mock_mqtt_component)
+from tests.components.vacuum import common
+
+COMMAND_TOPIC = 'vacuum/command'
+SEND_COMMAND_TOPIC = 'vacuum/send_command'
+STATE_TOPIC = 'vacuum/state'
+
+DEFAULT_CONFIG = {
+    CONF_PLATFORM: 'mqtt',
+    CONF_COMPONENT: 'state',
+    CONF_NAME: 'mqtttest',
+    CONF_COMMAND_TOPIC: COMMAND_TOPIC,
+    mqttvacuum.CONF_SEND_COMMAND_TOPIC: SEND_COMMAND_TOPIC,
+    CONF_STATE_TOPIC: STATE_TOPIC,
+    mqttvacuum.CONF_SET_FAN_SPEED_TOPIC: 'vacuum/set_fan_speed',
+    mqttvacuum.CONF_FAN_SPEED_LIST: ['min', 'medium', 'high', 'max'],
+}
+
+
+@pytest.fixture
+def mock_publish(hass):
```

Comment:
Why can't we use the existing `mqtt_mock` fixture that all mqtt tests have available?

Reply:
It was copy paste from old vacuum tests. Deleted it.